hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcClient.java

```diff
@@ -519,6 +525,21 @@ private Object invokeMethod(
     }
   }
 
+  /**
+   * For Tracking which is the actual client address.
+   * It adds key/value (clientIp/"ip") pair to the caller context.
+   */
+  private void appendClientIp2CallerContext() {
+    final CallerContext ctx = CallerContext.getCurrent();
+    CallerContext.Builder builder;
```

Comment:
Do this in the moment you create it in line 537.

Reply:
Ok